### PR TITLE
Support user pupil detection plugins

### PR DESCRIPTION
**Pull request by romanroibu**

### CODE REVIEW

File: pupil_src/launchables/eye.py [hunk 1]
```diff
@@ -202,8 +202,30 @@ def get_timestamp():
...
         g_pool.get_now = get_time_monotonic
 
+        def load_runtime_pupil_detection_plugins():
+            from plugin import import_runtime_plugins
+            from pupil_detector_plugins.detector_base_plugin import PupilDetectorPlugin
+
+            plugins_path = os.path.join(g_pool.user_dir, "plugins")
+
+            for plugin in import_runtime_plugins(plugins_path):
+                if not isinstance(plugin, type(PupilDetectorPlugin)):
```
Comment (papr) [inline pupil_src/launchables/eye.py:212]: The result of `type(PupilDetectorPlugin)` is `<class 'type'>`. I think, if you wanted to test for actual classes, this change might make your intent more clear:
```suggestion
                if not isinstance(plugin, type):
```

File: pupil_src/launchables/eye.py [hunk 2]
```diff
@@ -655,6 +685,17 @@ def window_should_update():
...
+                    subject.startswith("stop_eye_plugin")
+                    and notification["target"] == g_pool.process
+                ):
+                    try:
+                        plugin_to_stop = g_pool.plugin_by_name[notification["name"]]
+                    except KeyError as err:
+                        logger.error(f"Attempt to load unknown plugin: {err}")
+                    else:
+                        plugin_to_stop.alive = False
+                        g_pool.plugins.clean()
```
Comment (papr) [inline pupil_src/launchables/eye.py:698]: Please extract this part into a separate PR as a solution for #2035

**Review by papr — CHANGES_REQUESTED** [review]

File: pupil_src/launchables/eye.py (hunk 2 above)
Comment (romanroibu) [reply pupil_src/launchables/eye.py:698]: Opened PR https://github.com/pupil-labs/pupil/pull/2039

**Review by papr — APPROVED** [review]
